src/com/carolinarollergirls/scoreboard/jetty/WS.java

```diff
@@ -266,7 +265,54 @@ private void sendWSUpdatesForPaths(Set<String> watchedPaths, Set<String> changed
         }
 
         protected UUID id;
-        protected Set<String> paths = new HashSet<>();
+        protected PathTrie paths = new PathTrie();
         private Map<String, Object> state;
+
+        private class PathTrie {
+          boolean exists = false;
+          Map<Character, PathTrie> trie = new HashMap<>();
+
+          public void addAll(Set<String> c) {
+            for (String p: c) {
+              add(p);
+            }
+          }
+          public void add(String p) {
+            PathTrie head = this;
+            for (int i = 0;; i++) {
+              if (head.exists) {
+                // Already covered.
+                return;
+              }
+              if (i >= p.length()) {
+                break;
```

Comment:
This condition should be in the head of the loop.

Reply:
That'd result in us iterating once when we don't need to, which is why the head.exists check is before it.